runtime/lua/vim/fs.lua

```diff
@@ -112,14 +112,15 @@ function M.find(names, opts)
   names = type(names) == 'string' and { names } or names
 
   local path = opts.path or vim.loop.cwd()
+  local path_sep = vim.loop.os_uname().version:match('Windows') and '\\' or '/'
```

Comment:
You can use [package.config](https://www.lua.org/manual/5.2/manual.html#pdf-package.config) to get the path separator:
```suggestion
  local path_sep = package.config:sub(1, 1)
```

Reply:
we have `vim.fn.windowsversion()` function.

Reply:
shouldn't this respect the windows' option `shellslash` ?

Reply:
> shouldn't this respect the windows' option `shellslash` ?

Indeed it should. That complicates things a bit, but this should work:

```lua
    local path_sep = '/'
    if vim.fn.windowsversion() ~= '' and not vim.o.shellslash then
       path_sep = '\\'
    end
```

Reply:
it's `#vim.fn.windowsversion() > 0 `

Reply:
> it's `#vim.fn.windowsversion() > 0`

I figured checking if `shellslash` exists would be better anyway.

Reply:
use the `windowsversion` to check is enough I think

Reply:
> shouldn't this respect the windows' option `shellslash` ?

I don't think libuv respect `shellslash` tho? I'm getting the same results for `vim.loop.cwd()` either way

```vim
:set shellslash
:lua =vim.loop.cwd() " -> 'C:\\projects`
:set noshellslash
:lua =vim.loop.cwd() " -> still 'C:\\projects`
```

so what's the point in checking that when `libuv` isn't affected in the first place?